app/lib/project/MetapolatorProject.js

```diff
@@ -509,8 +509,54 @@ define([
         return this._controller;
     };
 
-    _p.import = function(masterName, sourceUFODir, glyphs) {
-        var importer = new ImportController( this._log, this,
+    _p.importZippedUFOInstances = function(filename, blob) {
+        var mem_io = new InMemory()
+          , imported_instances = Array()
+          ;
+
+        zipUtil.unpack(false, blob, mem_io, "");
+
+        var dirs = mem_io.readDir(false, "/")
+          , baseDir = dirs[0]
+          , names = mem_io.readDir(false, baseDir)
+          , n, l
+          ;
+
+        console.log("dirs:", dirs);
+
+        for (n=0, l=names.length; n<l; n++){
+            var name = names[n]
+              , UFOZip = baseDir + name //This may be wrong in some cases. We need a more robust implementation.
```

Comment:
On 13 May 2015 2:36 am, "Lasse Fister" <notifications@github.com> wrote:
>
> In app/lib/project/MetapolatorProject.js:
>
> > +          , imported_instances = Array()
> > +          ;
> > +
> > +        zipUtil.unpack(false, blob, mem_io, "");
> > +
> > +        var dirs = mem_io.readDir(false, "/")
> > +          , baseDir = dirs[0]
> > +          , names = mem_io.readDir(false, baseDir)
> > +          , n, l
> > +          ;
> > +
> > +        console.log("dirs:", dirs);
> > +
> > +        for (n=0, l=names.length; n<l; n++){
> > +            var name = names[n]
> > +              , UFOZip = baseDir + name //This may be wrong in some
cases. We need a more robust implementation.
>
> So our initial model for uploaded files is like:
>
> upload.zip
>     ├── master1.ufo.zip
>     │    └── master1.ufo
>     ├── master2.ufo.zip
>     │    └── master2.ufo
>     └── master3.ufo.zip
>          └── master3.ufo
>
> And not just:
>
> upload.zip
>     ├── master1.ufo
>     ├── master2.ufo
>     └── master3.ufo
>
> @davelab6 does this sound ok?

Yes, it mirrors the export.

> I would think that we'd want to have other schemes than the former one
immediately.

I'm OK with hoops for 0.3.0 as it sets an agenda for paying users

> Also, could you please document the expected scheme above the function as
in
>
> /**
>  * Put here a description of the blob argument especially the expected
file structure.
>  * Maybe a `tree` like diagram like mine above to illustrate it:
>  * upload.zip
>  *     ├── master1.ufo.zip
>  *     │    └── master1.ufo
>  *     ├── master2.ufo.zip
>  *     │    └── master2.ufo
>  *     └── master3.ufo.zip
>  *          └── master3.ufo
>  */
> _p.importZippedUFOInstances = function(blob) {}
>
> —
> Reply to this email directly or view it on GitHub.